### PR TITLE
etcdserver/api: remove capnslog

### PR DESCRIPTION
**Pull request by jingyih**
Remove dependency of `capnslog` in pkgs under `etcdserver/api`.

Related: #11426

### CODE REVIEW

File: etcdserver/api/v2http/httptypes/errors.go [hunk 1]
```diff
@@ -40,7 +35,7 @@ func (e HTTPError) WriteTo(w http.ResponseWriter) error {
 	w.WriteHeader(e.Code)
 	b, err := json.Marshal(e)
 	if err != nil {
-		plog.Panicf("marshal HTTPError should never fail (%v)", err)
+		panic(fmt.Sprintf("failed to marshal HTTPError: %v", err))
```
Comment (jingyih) [inline etcdserver/api/v2http/httptypes/errors.go]: Note: not sure how to pass in zap Logger here.

File: etcdserver/api/v3rpc/header.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func newHeader(s *etcdserver.EtcdServer) header {
 // fill populates pb.ResponseHeader using etcdserver information
 func (h *header) fill(rh *pb.ResponseHeader) {
 	if rh == nil {
-		plog.Panic("unexpected nil resp.Header")
+		panic("unexpected nil resp.Header")
```
Comment (jingyih) [inline etcdserver/api/v3rpc/header.go]: Note: same here, not sure how to pass in zap Logger.

**Comment by jingyih** [comment]
This PR is ready to be reviewed. Please help take a look if you have bandwidth. Thanks! @gyuho @xiang90 @spzala

File: etcdserver/api/v3rpc/header.go (hunk 2 above)
Comment (gyuho) [reply etcdserver/api/v3rpc/header.go]: Think just `panic` is fine.

**Comment by gyuho** [comment]
Can we resolve git conflicts

**Comment by jingyih** [comment]
> Can we resolve git conflicts

Conflicted due to changes made in #11608. Rebased to current master HEAD.

**Pull request merged**